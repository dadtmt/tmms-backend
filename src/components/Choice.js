import React, { PropTypes } from 'react'
import { ListGroupItem } from 'react-bootstrap'

import RenderContent from '../renderers/RenderContent'
import RichTextDisplay from './RichTextDisplay'

const Choice = ({ content, made, text, type }) => <ListGroupItem active={made}>
  <RichTextDisplay rawContent={text} />
  <RenderContent made={made} type={type} {...content} />
</ListGroupItem>

Choice.propTypes = {
  content: PropTypes.object,
  made: PropTypes.bool.isRequired,
  text: PropTypes.object.isRequired,
  type: PropTypes.string
}

export default Choice
