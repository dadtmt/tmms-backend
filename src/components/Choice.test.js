import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Choice from './Choice'

const text = {
  blocks: [
    {
      data: {},
      depth: 0,
      entityRanges: [],
      inlineStyleRanges: [],
      key: '4bkmd',
      text: 'some rich text',
      type: 'unstyled'
    }
  ],
  entityMap: {}
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  const props = {
    id: 'SOME_ID',
    made: false,
    text
  }
  ReactDOM.render(
    <Choice {...props} />,
    div
  )
})

jest.mock('draft-js/lib/generateRandomKey', () => () => '123')

it('renders correctly with default type', () => {
  const props = {
    id: 'SOME_ID',
    made: false,
    text,
    type: 'default'
  }
  expect(
    renderer.create(<Choice {...props} />).toJSON()
  ).toMatchSnapshot()
})

it('renders correctly with dice type', () => {
  const props = {
    content: {
      details: 'details',
      master: false,
      modifier: 0,
      nbDices: 1,
      nbSides: 6,
      result: 6
    },
    id: 'SOME_ID',
    made: false,
    text,
    type: 'dice'
  }
  expect(
    renderer.create(<Choice {...props} />).toJSON()
  ).toMatchSnapshot()
})
