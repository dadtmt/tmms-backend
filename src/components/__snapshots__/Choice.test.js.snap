exports[`test renders correctly with default type 1`] = `
<span
  className="list-group-item">
  <div
    className="DraftEditor-root">
    <div
      className="DraftEditor-editorContainer">
      <div
        aria-activedescendant={null}
        aria-autocomplete={null}
        aria-describedby={null}
        aria-expanded={null}
        aria-haspopup={null}
        aria-label={undefined}
        aria-owns={null}
        className="public-DraftEditor-content"
        contentEditable={false}
        data-testid={undefined}
        onBeforeInput={[Function]}
        onBlur={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onCopy={[Function]}
        onCut={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseUp={[Function]}
        onPaste={[Function]}
        onSelect={[Function]}
        role={null}
        spellCheck={false}
        style={
          Object {
            "outline": "none",
            "whiteSpace": "pre-wrap",
            "wordWrap": "break-word",
          }
        }
        suppressContentEditableWarning={true}
        tabIndex={undefined}>
        <div
          data-contents="true">
          <div
            className=""
            data-block={true}
            data-editor="123"
            data-offset-key="4bkmd-0-0">
            <div
              className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
              data-offset-key="4bkmd-0-0">
              <span
                data-offset-key="4bkmd-0-0"
                style={Object {}}>
                <span
                  data-text="true">
                  some rich text
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</span>
`;

exports[`test renders correctly with dice type 1`] = `
<span
  className="list-group-item">
  <div
    className="DraftEditor-root">
    <div
      className="DraftEditor-editorContainer">
      <div
        aria-activedescendant={null}
        aria-autocomplete={null}
        aria-describedby={null}
        aria-expanded={null}
        aria-haspopup={null}
        aria-label={undefined}
        aria-owns={null}
        className="public-DraftEditor-content"
        contentEditable={false}
        data-testid={undefined}
        onBeforeInput={[Function]}
        onBlur={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onCopy={[Function]}
        onCut={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseUp={[Function]}
        onPaste={[Function]}
        onSelect={[Function]}
        role={null}
        spellCheck={false}
        style={
          Object {
            "outline": "none",
            "whiteSpace": "pre-wrap",
            "wordWrap": "break-word",
          }
        }
        suppressContentEditableWarning={true}
        tabIndex={undefined}>
        <div
          data-contents="true">
          <div
            className=""
            data-block={true}
            data-editor="123"
            data-offset-key="4bkmd-0-0">
            <div
              className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
              data-offset-key="4bkmd-0-0">
              <span
                data-offset-key="4bkmd-0-0"
                style={Object {}}>
                <span
                  data-text="true">
                  some rich text
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span
      className="label label-default">
      Lancer les d√©s: 
      1D6+0
    </span>
  </div>
</span>
`;
